src/librustc/ty/layout.rs

```diff
@@ -1762,6 +1771,125 @@ impl<'a, 'tcx, C> TyLayoutMethods<'tcx, C> for Ty<'tcx>
             }
         })
     }
+
+    fn pointee_info_at(this: TyLayout<'tcx>, cx: &C, offset: Size
+    ) -> Option<PointeeInfo> {
+        let mut result = None;
```

Comment:
I think you should not have a mutable variable, but instead just do an early return wherever you are currently assigning to `result` and return `None` at the end if no early return happened

Reply:
Agreed, will change this :+1: